py-server/default_model.py

```diff
@@ -23,37 +24,38 @@
 """
 
 
-class DefaultModel:
+class DefaultAgent(ConversationalAgent):
     def __init__(self):
-
         start_time = time.time()
         self.chat_model = ChatModel.from_pretrained("chat-bison@002")
         end_time = time.time()
-        print(f"Created DefaultModel in: {end_time - start_time:.2f} seconds")
+        print(f"Created DefaultAgent in: {end_time - start_time:.2f} seconds")
 
-    def chat(self, message_history, world_state, message):
+    def chat(self, message: str):
         start_time = time.time()
-        messages = self.convert_message_history(message_history)
         chat_session = self.chat_model.start_chat(
-            context=PROMPT, message_history=messages
+            context=PROMPT, message_history=self._state.message_history
         )
         res = chat_session.send_message(message)
         end_time = time.time()
-        print(f"DefaultModel chat took: {end_time - start_time:.2f} seconds")
+        print(f"DefaultAgent chat took: {end_time - start_time:.2f} seconds")
         text = res.candidates[0].text
-        return text, None
+        return text
 
-    def convert_message_history(self, message_history):
+    def _build_message_history(self, state):
         """
         Converts langchain message history to ChatMessage format
         so that we can always refer back to the direct API version.
         """
+        if not state.message_history:
+            return None
         messages = []
-        for message in message_history:
+        for message in state.message_history:
```

Comment:
I wonder if this is the right time to ditch the direct API version and use langchain throughout so we don't need to do this message conversion anymore?

Reply:
removed in fa40e6f